### PR TITLE
Cleanup

### LINKED ISSUE #115: CI doesn't seems to not actually run for MSRV
When setting up CI for axum I just copied the setup from Tower thinking that would be fine. However I didn't notice it included an MSRV of 1.40 [which is tested on CI](https://github.com/tokio-rs/axum/blob/main/.github/workflows/CI.yml#L15).

axum requires at least Rust 1.51 due the use of const generics in [`ContentLengthLimit`](https://github.com/tokio-rs/axum/blob/main/src/extract/content_length_limit.rs#L24). So it seems the 1.40 step on CI isn't actually running, since it should be failing 🤔 

We should fix CI so it actually does perform the checks and bump our MSRV to 1.51. If it turns out we use a feature not available on 1.51 I would be fine with requiring an even newer version, but it depends on the exact case.

### PR DESCRIPTION
**Pull request by jplatte**
Commit descriptions should explain all changes :)

Fixes #115.

### CODE REVIEW

**Comment by jplatte** [comment]
This now removes one check from CI (second-to-last commit) but should be better than the status quo (due to #115).

**Comment by davidpdrsn** [comment]
`create_url_params` is only called in tests. So changing the call sites to use a slice is fine.

Thanks for doing this by the way! Gonna review it over the next few days.

File: Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,6 @@
 authors = ["David Pedersen <david.pdrsn@gmail.com>"]
 categories = ["asynchronous", "network-programming", "web-programming"]
 description = "Web framework that focuses on ergonomics and modularity"
-documentation = "https://docs.rs/axum/0.1.3"
```
Comment (taiki-e) [inline Cargo.toml:5]: docs.rs selects the latest version automatically, but I don't think crate.io sets the docs URL automatically. ([example](https://crates.io/crates/ruma))

```diff
+ documentation = "https://docs.rs/axum"
```

Comment (jplatte) [reply Cargo.toml:5]: I think you're blocking the script from docs.rs that adds the link on crates.io :)

Comment (taiki-e) [reply Cargo.toml:5]: Oh, sorry! You are right.

**Comment by davidpdrsn** [comment]
I'm gonna merge this and fix CI on main. Would like to have this in when I start merging some of the many open PRs.

**Pull request merged**